### PR TITLE
Use SupportsFeatureMixin to interrogate a vm/template

### PR DESCRIPTION
**Pull request by jameswnl**

### CODE REVIEW

File: app/models/manageiq/providers/cloud_manager/vm.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@ def resize_confirm
     raw_resize_confirm
   end
 
-  def resize_revert
+  def resize_rever
```
Comment (h-kataria) [inline app/models/manageiq/providers/cloud_manager/vm.rb]: @jameswnl why is this method being renamed, is that a typo?

File: app/models/vm_or_template.rb [hunk 2]
```diff
@@ -30,9 +30,8 @@ class VmOrTemplate < ApplicationRecord
   include TenancyMixin
 
   include AvailabilityMixin
-  include SupportsFeatureMixin
 
-  supports_not :retirement, ""
+  supports_not :retirement
```
Comment (h-kataria) [inline app/models/vm_or_template.rb]: @jameswnl making this change seems to be throwing an error for me
``
/home/hkataria/dev/manageiq/app/models/mixins/supports_feature_mixin.rb:77:in `supports_not': wrong number of arguments (1 for 2) (ArgumentError)
	from /home/hkataria/dev/manageiq/app/models/vm_or_template.rb:34:in `<class:VmOrTemplate>'
	from /home/hkataria/dev/manageiq/app/models/vm_or_template.rb:5:in `<top (required)>'
``

File: app/models/manageiq/providers/cloud_manager/vm.rb (hunk 1 above)
Comment (jameswnl) [inline app/models/manageiq/providers/cloud_manager/vm.rb]: sorry, that was a typo when working with Gaurav. I've fixed that now.

File: app/models/vm_or_template.rb (hunk 2 above)
Comment (jameswnl) [inline app/models/vm_or_template.rb]: Did you pull in the latest changes from @durandom?

Comment (durandom) [inline app/models/vm_or_template.rb]: Yeah, you have to rebase on master first.
Sorry, for the trouble @h-kataria - it's one of the first applications of this so it's a bit of a rough ride...

**Pull request closed without merging**